### PR TITLE
Limit execution errors printed per file

### PR DESCRIPTION
**Pull request by i-maravic**

### CODE REVIEW

**Comment by HenryCaiHaiying** [comment]
Can you sync to the latest code?

We have changed those code quite a bit (especially on error handling and test framework side), you can also take a look at CamusJobTestWithMock to see how to do those unit testing.

**Comment by i-maravic** [comment]
@HenryCaiHaiying When you say to sync this with latest code what do you mean? This is already based on the latest master.

CamusJobTestWithMock is already testing the most important feature of the function checkExecutionErrors(), ie. if there was errors or not.

Since I don't see error logging as critical part of Camus I'm a bit reluctant to add test for them.

What do you think?

File: camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java [hunk 1]
```diff
@@ -476,20 +464,40 @@ public void cancel() throws IOException {
...
+
+        final List<Pair<EtlKey, ExceptionWritable>> errorsFromFile =
+            errors.containsKey(filePath.toString()) ?
+            errors.get(filePath.toString()) :
+            Lists.<Pair<EtlKey, ExceptionWritable>>newArrayList();
+
+        if (errorCounter >= maxErrorsFromFile) {
+          errorsFromFile.add(
+              new Pair<EtlKey, ExceptionWritable>(new EtlKey(key),
+                                                  new ExceptionWritable("... Too many errors to show. Skipping...")));
```
Comment (zliu41) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java]: It may be better to include the number of errors skipped.

Comment (i-maravic) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java]: Done.

**Comment by zliu41** [comment]
@i-maravic Thanks.

**Pull request merged**